src/libstore/filetransfer.cc

```diff
@@ -1004,4 +1147,28 @@ FileTransferError::FileTransferError(
         err.msg = hf;
 }
 
+#if NIX_WITH_S3_SUPPORT
+void curlFileTransfer::clearCredentialCache()
+{
+    auto cache(credentialProviderCache.lock());
+    cache->providers.clear();
+}
+
+// Function called by AWS CRT shutdown to clear credential provider cache
+void cleanupCredentialProviderCache()
+{
+    // Access the curlFileTransfer singleton and clear its cache
+    try {
+        auto ft = getFileTransfer();
```

Comment:
This will not work as expected in builtin derivaiton fetching, because it doesn't use `getFileTransfer` in that case because it is in a fork.

Reply:
Yeah, those would each end up with their own provider. I couldn't see a clean way to improve that though, and AFAICT the status-quo also does that.

Reply:
We should add a test though.